src/codegen/codegen_cpp_visitor.hpp

```diff
@@ -1500,6 +1500,12 @@ class CodegenCppVisitor: public visitor::ConstAstVisitor {
      */
     void print_table_check_function(const ast::Block&);
 
+    /**
+     * Whether or not we need to emit code for CVODE (only makes a difference
+     * for NEURON codegen).
+     */
+    bool emit_cvode() const;
```

Comment:
Should this be part of the CppVisitor if we only need it for NEURON? Do we need it for anything other than testing? If no, is there a reasonably simple change prevents adding a new getter?

Reply:
> Do we need it for anything other than testing?

No, which is why I was slightly annoyed when I added it as a shortcut, but I'm not really sure how else to do it without a new getter (other than making the whole `CodegenInfo` struct `public`). Ideas are welcome!

Reply:
Since you only need the `info` object, not the visitor, can you create the `CodegenInfo` on it's own? Not as part of the visitor, just by itself?

If not, please review why not. Then we can create the appropriate issue :)

Reply:
The `codegen_helper` tests provided some hints on how to achieve this without changing the API, so the new ones for CVODE-related codegen are now there (which IMHO makes more sense since the changes in the PR are actually to that visitor instead of the NEURON codegen one).

Reply:
You're correct, it makes more sense to test `info` directly.